components/allocator/lib.rs

```diff
@@ -39,7 +39,10 @@ mod platform {
         #[cfg(target_os = "linux")]
         return libc::malloc_usable_size(ptr as *mut _);
 
-        #[cfg(not(target_os = "linux"))]
+        #[cfg(target_vendor = "apple")]
+        return libc::malloc_size(ptr);
+
+        #[cfg(not(any(target_os = "linux", target_vendor = "apple")))]
```

Comment:
Can you also try to combine the two code paths for linux and android. The CI should catch it if it's an issue? So this would be:

```rust
        #[cfg(not(target_vendor = "apple")))]
        return libc::malloc_usable_size(ptr as *mut _);

        #[cfg(target_vendor = "apple")]
        return libc::malloc_size(ptr);
```

Reply:
I didn't know that rust can make *mut ptr to *const ptr automagically.